### ISSUE TITLE
Oracle support?

### ISSUE THREAD

**Issue by piotr-dobrogost** [kvesteri/postgresql-audit#15]
Would Oracle support be in scope of this project?
Has any work been already done in this direction?

**Comment by kvesteri** [comment]
Yes definitely, technically it should be possible to implement this if Oracle has a JSON datatype.

**Comment by piotr-dobrogost** [comment]
From _[Did You Say "JSON Support" in Oracle 12.1.0.2?](https://blogs.oracle.com/OTN-DBA-DEV-Watercooler/entry/did_you_say_json_support)_
> So for these reasons (and maybe a few more) Oracle Database 12c supports JSON natively with relational database features, including transactions, indexing, declarative querying, and views. – 

Also [JSON in Oracle Database](https://docs.oracle.com/database/121/ADXDB/json.htm#ADXDB6246) in Oracle's documentation.

There are some extensions adding support for JSON for earlier versions of Oracle. To know if they would work one needs to know specific requirements – what is required specifically with regard to JSON support for this project to work?

**Comment by kvesteri** [comment]
You can browse through the SQL templates to see what needs to supported by Oracle as well: https://github.com/kvesteri/postgresql-audit/tree/master/postgresql_audit/templates

Basically we would need Oracle equivalents for those SQL templates.
